Kernel/Term.cpp

```diff
@@ -776,7 +774,7 @@ const vstring& Literal::predicateName() const
 
 
 bool Literal::isAnswerLiteral() const {
-  return env.signature->getPredicate(functor())->answerPredicate();
+  return isNegative() && env.signature->getPredicate(functor())->answerPredicate();
```

Comment:
Are answer literals *always* negative? When @hzzv gets to it she might want to check this against her synthesis code.

Reply:
A good point to ask Petra to check.

The reason I added this is that, while normally the added answer literal really occurs only negatively (except in the answer literal resolver, the unit clause which helps closing the proof, but never enters proof search explicitly) I saw it positive under AVATAR (through add_complementary) and the clause were it occurred was not meant for the answer literal resolver to resolve against and finish the proof.

Reply:
In synthesis the answer literals should always be negative.
If you're adding some logic that also allows them in positive context, maybe we could also add an assert somewhere checking that if `QuestionAnsweringMode::SYNTHESIS` is on, the answer literals are negative? (Maybe in `UncomputableAnswerLiteralRemoval::simplify`.)